### PR TITLE
fix(terraform): change CKV_AWS_33 into CKV2 check

### PR DESCRIPTION
**Pull request by Michael-McClelland**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description

~An error occurs in check CKV_AWS_33 when the key policy is large.  The dictionary that the check passes to the for loop becomes a string once a certain line count threshold is reached.  Since no exception is thrown as the loop processes the string the check produces no error and silently and incorrectly passes.~

Converts CKV_AWS_33 into V2 formatted check

~Fixes~ (#5627)

### Fix
~When executed in the new graph check format format a check similar to CKV_AWS_33 produces no error.  Thus the simplest correction is to upgrade the check to the new format.~

Upgrades check to graph format method.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/terraform/checks/graph_checks/aws/KmsKeyWildcardPrincipal.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      resource_types:
+            - "aws_kms_key_policy"
+      attribute: "policy.Statement[?(@.Effect == 'Allow' & @.Principal == '*')]"
+      operator: "jsonpath_not_exists"
+    - cond_type: "attribute"
+      resource_types:
+            - "aws_kms_key_policy"
+      attribute: "policy.Statement[?(@.Effect == 'Allow')].Principal.AWS[*]"
+      operator: "jsonpath_not_equals"
+      value: "*"
```
Comment (gruebel) [inline checkov/terraform/checks/graph_checks/aws/KmsKeyWildcardPrincipal.yaml:28]: your policy is handling key to policy connection, but not when the policy is inlined, that's why the actual test fails for all resources.

File: checkov/terraform/checks/graph_checks/aws/KmsKeyWildcardPrincipal.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+---
+metadata:
+  id: "CKV2_AWS_67"
```
Comment (gruebel) [inline checkov/terraform/checks/graph_checks/aws/KmsKeyWildcardPrincipal.yaml:3]:
```suggestion
  id: "CKV_AWS_33"
```
no need to create a new ID

File: checkov/terraform/checks/graph_checks/aws/KmsKeyWildcardPrincipal.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  name: "Ensure KMS key policy does not contain wildcard (*) principal"
+  category: "GENERAL_SECURITY"
```
Comment (gruebel) [inline checkov/terraform/checks/graph_checks/aws/KmsKeyWildcardPrincipal.yaml:5]:
```suggestion
  category: "IAM"
```
this fits best

**Review by gruebel — COMMENTED** [review]
I think it is a good idea to migrate it over to a YAML based policy to be able to cover the case of key to policy connection 🏅

File: checkov/terraform/checks/graph_checks/aws/KmsKeyWildcardPrincipal.yaml (hunk 2 above)
Comment (Michael-McClelland) [reply checkov/terraform/checks/graph_checks/aws/KmsKeyWildcardPrincipal.yaml]: Id change made

File: checkov/terraform/checks/graph_checks/aws/KmsKeyWildcardPrincipal.yaml (hunk 3 above)
Comment (Michael-McClelland) [reply checkov/terraform/checks/graph_checks/aws/KmsKeyWildcardPrincipal.yaml]: category change set

**Comment by Michael-McClelland** [comment]
There is a bug in this I cannot explain.  When evaluating keys that use a data block for the policy unless the resources involved trigger a fail from the check it doesn't report the resource as having been evaluated.  Passes are not reported when the key policy is coming from a data resource, only failures.  

For example this reports a non-compliant resource as intended

```
provider "aws" {
  region = "us-east-1"
}

data "aws_iam_session_context" "current" {
  arn = data.aws_caller_identity.current.arn
}
data "aws_caller_identity" "current" {}
data "aws_organizations_organization" "current" {}
data "aws_partition" "current" {}
data "aws_region" "current" {}

data "aws_iam_policy_document" "keypolicy" {

  statement {
    principals {
      type = "AWS"
      identifiers = [
        "*"
      ]
    }
    effect = "Allow"
    actions = [
      "kms:*"
    ]
    resources = [
      "*",
    ]
  }

}


resource "aws_kms_key" "TestKey" {
  description                        = "testkey"
  deletion_window_in_days            = 7
  key_usage                          = "ENCRYPT_DECRYPT"
  customer_master_key_spec           = "SYMMETRIC_DEFAULT"
  policy                             = data.aws_iam_policy_document.keypolicy.json
  bypass_policy_lockout_safety_check = false
  is_enabled                         = true
  enable_key_rotation                = true
}
```

but this silently passes and is not acknowledged as compliant
```
provider "aws" {
  region = "us-east-1"
}


data "aws_iam_session_context" "current" {
  arn = data.aws_caller_identity.current.arn
}
data "aws_caller_identity" "current" {}
data "aws_organizations_organization" "current" {}
data "aws_partition" "current" {}
data "aws_region" "current" {}

data "aws_iam_policy_document" "keypolicy" {

  statement {
    principals {
      type = "AWS"
      identifiers = [
        "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.id}:root"
      ]
    }
    effect = "Allow"
    actions = [
      "kms:*"
    ]
    resources = [
      "*",
    ]
  }

}


resource "aws_kms_key" "TestKey" {
  description                        = "testkey"
  deletion_window_in_days            = 7
  key_usage                          = "ENCRYPT_DECRYPT"
  customer_master_key_spec           = "SYMMETRIC_DEFAULT"
  policy                             = data.aws_iam_policy_document.keypolicy.json
  bypass_policy_lockout_safety_check = false
  is_enabled                         = true
  enable_key_rotation                = true
}
```

**Comment by Saarett** [comment]
@Michael-McClelland How do you feel about @gruebel 's [comment](https://github.com/bridgecrewio/checkov/issues/5627#issuecomment-1761821012)? I tend to agree with him

**Pull request closed without merging**